File: 637-average-of-levels-in-binary-tree/637-average-of-levels-in-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Double> averageOfLevels(TreeNode root) {
        List<Double> toReturn = new ArrayList<Double>();
        Queue<TreeNode> q = new LinkedList<>();
        q.offer(root);

        while(!q.isEmpty()) {
        	int size = q.size();
        	double calcValue = 0;
        
        	for(int i = 0; i < size; i++) {
            	TreeNode currentNode = q.poll();
            	calcValue += currentNode.val;
            	
				if (currentNode.left != null)
					q.add(currentNode.left);
				if (currentNode.right != null)
					q.add(currentNode.right);
        	}
        	toReturn.add(calcValue / size);
        }
        return toReturn;
    }
}

File: 637-average-of-levels-in-binary-tree/637-average-of-levels-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def averageOfLevels(self, root: Optional[TreeNode]) -> List[float]:
        if not root:
            return 0

        queue = []
        queue.append(root)

        levelAvg = []

        while queue:
            sum = 0
            size = len(queue)

            for i in range(size):
                currentElement = queue.pop(0)
                sum += currentElement.val

                if currentElement.left:
                    queue.append(currentElement.left)
                if currentElement.right:
                    queue.append(currentElement.right)

            levelAvg.append(sum / size)

        return levelAvg